providers/providers.yaml

```diff
@@ -0,0 +1,28 @@
+# This file contains the configuration for supported providers.
+
+# All provider blocks have two kinds of fields:
+# 1) Connector specific, which define properties specific to the go connector itself (e.g. connector_type)
+# 2) Provider specific, which define configuration related to the actual provider API (e.g. provider_base_url)
+# Each field is prefixed accordingly as a convenience and convention.
+
+# This YAML file allows for {{.variables}} (note the period) in the configuration, which can be replaced at runtime
+# by the connector. This syntax (and more) comes from the https://pkg.go.dev/html/template package.
+
+providers:
+  salesforce:
+    connector_type: full
+    connector_version: 1.0.0
+    provider_auth_type: oauth2
+    provider_base_url: https://{{.subdomain}}.salesforce.com
```

Comment:
Can we use `workspace` instead of `subdomain` as the term here? Workspace is a more generic term, whereas subdomain is very Salesforce-specific. We also use `workspace` in our data schemas already (e.g. the Connections table has a `providerWorkspaceRef`)

Reply:
This is meant to be salesforce specific. So when the salesforce client will read config, it will look like - 
```
params = &salesforceParams{
  subdomain: ..,
}

configMap, err := config.Read(providers.Salesforce, map[string]string{
  "subdomain": params.subdomain,
})

params.base_url = configMap["provider_base_url"]
```

The substitutions are done per provider, so we should use the same term that we are using in the params and inputs for ease of use / understanding